frontend/src/services/courses-api/courses-api.service.ts

```diff
@@ -31,6 +31,18 @@ class Courses {
       },
     );
   }
+
+  public getByName(name: string): Promise<CourseGetResponseDto[]> {
+    return this.#http.load(
+      `${this.#apiPrefix}${ApiPath.COURSES}${
+        CoursesApiPath.ROOT
+      }?title=${name}`,
```

Comment:
```suggestion
      }?=${name}`,
```
we need to provide an object of params